usvm-ts/src/main/kotlin/org/usvm/machine/TsInterpreterObserver.kt

```diff
@@ -0,0 +1,85 @@
+package org.usvm.machine
+
+import org.jacodb.ets.base.EtsAssignStmt
+import org.jacodb.ets.base.EtsCallExpr
+import org.jacodb.ets.base.EtsGotoStmt
+import org.jacodb.ets.base.EtsIfStmt
+import org.jacodb.ets.base.EtsReturnStmt
+import org.jacodb.ets.base.EtsSwitchStmt
+import org.jacodb.ets.base.EtsThrowStmt
+import org.usvm.UBoolExpr
+import org.usvm.machine.expr.TsSimpleValueResolver
+import org.usvm.machine.interpreter.TsStepScope
+import org.usvm.statistics.UInterpreterObserver
+
+@Suppress("unused")
+interface TsInterpreterObserver : UInterpreterObserver {
+    fun onAssignStatement(
+        simpleValueResolver: TsSimpleValueResolver,
+        stmt: EtsAssignStmt,
+        scope: TsStepScope,
+    ) {
+        // default empty implementation
+    }
+
+    // TODO on entry point
+
+    fun onCallWithUnresolvedArguments(
+        simpleValueResolver: TsSimpleValueResolver,
+        stmt: EtsCallExpr,
+        scope: TsStepScope,
+    ) {
+        // default empty implementation
+    }
+
+    // TODO onCallWithResolvedArguments
```

Comment:
TODO

Reply:
We don't have supported enough calls insts yet